### PR TITLE
Remove obsolete 'std' directory

### LINKED ISSUE #3662: Document the "std" directory
Document the existence of the files in the `std` directory and that they should only be referred to by the git version tag.

**or** remove them entirely.

### PR DESCRIPTION
**Pull request by axic**
Fixes #3662.

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Breaking Changes:
  * Type Checker: Disallow arithmetic operations for Boolean variables.
  * Disallow trailing dots that are not followed by a number.
  * Remove assembly instructions ``sha3`` and ``suicide``
+ * Removed the obsolete files in the ``std`` directory.
```
Comment (erak) [inline Changelog.md]: Should be ``remove`` :) Perhaps ``Remove obsolete std directory from project.`` is even better?

**Review by erak — CHANGES_REQUESTED** [review]

File: Changelog.md (hunk 1 above)
Comment (axic) [reply Changelog.md]: Is it better this way?

File: Changelog.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ Breaking Changes:
  * Type Checker: Disallow arithmetic operations for Boolean variables.
  * Disallow trailing dots that are not followed by a number.
  * Remove assembly instructions ``sha3`` and ``suicide``
+ * Remove obsolete ``std` directory from the Solidity repository. This means accessing ``https://github.com/etheruem/soldity/std/*.sol`` will not be possible.
```
Comment (erak) [inline Changelog.md]: There're some typos and missing back-ticks which breaks the markdown rendering. I'd go for (but fine with how it is): 

Remove obsolete ``std`` directory from the Solidity repository. Accessing ``https://github.com/ethereum/solidity/std/*.sol`` is not be possible any more.

**Review by erak — DISMISSED** [review]

File: test/cmdlineTests.sh [hunk 3]
```diff
@@ -113,14 +109,6 @@ do
 done
 )
 
-printTask "Compiling all files in std and examples..."
-
-for f in "$REPO_ROOT"/std/*.sol
-do
-    echo "$f"
-    compileWithoutWarning "$f"
```
Comment (axic) [inline test/cmdlineTests.sh:121]: It seems that we were running the compilation twice :)

File: Changelog.md [hunk 4]
```diff
@@ -11,6 +11,7 @@ Breaking Changes:
  * Type Checker: Disallow arithmetic operations for Boolean variables.
  * Disallow trailing dots that are not followed by a number.
  * Remove assembly instructions ``sha3`` and ``suicide``
+ * Remove obsolete ``std`` directory from the Solidity repository. This means accessing ``https://github.com/etheruem/soldity/std/*.sol`` will not be possible.
```
Comment (erak) [inline Changelog.md]: There're still typos in the URL ;)

Comment (axic) [reply Changelog.md]: Damn 😉

File: test/cmdlineTests.sh (hunk 3 above)
Comment (chriseth) [reply test/cmdlineTests.sh:121]: The first one actually checked that they produce bytecode, i.e. that the contracts are not abstract and that the bytecode generator works.

**Review by chriseth — APPROVED** [review]

**Pull request merged**